File: leet_SLWD_187_repeated_dna_sequence.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s)
    {
        int len = s.size();
        if (len < 10)
            return vector<string>();
        set<string> seen, r;
        int         i = -1;
        while (++i < len - 9)
        {
            string  temp = s.substr(i, 10);
            if (seen.find(temp) != seen.end())
                r.insert(temp);
            seen.insert(temp);
        }
        return vector<string>(r.begin(), r.end());
    }
};

File: leet_SLWD_187_repeated_dna_sequence.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        n = len(s)
        if n < 10:
            return []
        r, seen = set(), set()
        for i in range(n - 9):
            temp = s[i: i + 10]
            if temp in seen:
                r.add(temp)
            seen.add(temp)
        return list(r)
